### PR TITLE
[SPARK-37385][SQL][TESTS] Add tests for TimestampNTZ and TimestampLTZ for Parquet data source 

### PR DESCRIPTION
**Pull request by sadikovi**
### What changes were proposed in this pull request?

The PR updates/adds more tests for TimestampNTZ and TimestampLTZ types support in Parquet data source to make sure most of the cases are covered.

### Why are the changes needed?

Improves test coverage of the TimestampNTZ feature.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Existing unit tests.

### CODE REVIEW

**Comment by sadikovi** [comment]
cc @gengliangwang @cloud-fan

**Comment by HyukjinKwon** [comment]
cc @MaxGekk too

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala [hunk 1]
```diff
@@ -122,7 +122,63 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
   test("SPARK-36182: TimestampNTZ") {
     val data = Seq("2021-01-01T00:00:00", "1970-07-15T01:02:03.456789")
       .map(ts => Tuple1(LocalDateTime.parse(ts)))
-    checkParquetFile(data)
+    withAllParquetReaders {
+      checkParquetFile(data)
+    }
+  }
+
+  test("SC-88419: Read TimestampNTZ and TimestampLTZ for various logical TIMESTAMP types") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala:130]:
```suggestion
  test("Read TimestampNTZ and TimestampLTZ for various logical TIMESTAMP types") {
```

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by sadikovi** [comment]
Thank you!

**Pull request closed without merging**